### PR TITLE
chore(ci): Update CI workflows

### PR DESCRIPTION
**Pull request by jeqo**
Currently, all workflows run on every PR, causing duplicated executions. This PR fixes this, and separate tests into different steps to easier troubleshooting. 
Also, includes building docker image.

### CODE REVIEW

File: .github/workflows/pull_request_workflow.yml [hunk 1]
```diff
@@ -20,5 +20,12 @@ jobs:
         with:
           java-version: ${{ matrix.java-version }}
           distribution: temurin
+
       - name: Build with Gradle
-        run: ./gradlew build distTar
+        run: ./gradlew build distTar -x test
+
+      - name: Run integration tests
+        run: ./gradlew integrationTest
```
Comment (AnatolyPopov) [inline .github/workflows/pull_request_workflow.yml:28]: So we are skipping the unit tests entirely, right?

File: .github/workflows/pull_request_workflow.yml [hunk 2]
```diff
@@ -20,5 +20,12 @@ jobs:
         with:
           java-version: ${{ matrix.java-version }}
           distribution: temurin
+
       - name: Build with Gradle
-        run: ./gradlew build distTar
+        run: ./gradlew build distTar -x test
```
Comment (AnatolyPopov) [inline .github/workflows/pull_request_workflow.yml:25]: And do we continue running the integrationTests here despite the next stage?

File: .github/workflows/pull_request_workflow.yml (hunk 1 above)
Comment (jeqo) [reply .github/workflows/pull_request_workflow.yml:28]: My bad. Meant to disable integrationTest on previous state. Should be fixed now.

File: .github/workflows/main_push_workflow.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 name: Main checks after push
 on:
   push:
-    branches: [ '**' ]
+    branches: [ 'main' ]
```
Comment (AnatolyPopov) [inline .github/workflows/main_push_workflow.yml:5]: So we are not running the pipeline on push to branches anymore?

Comment (jeqo) [reply .github/workflows/main_push_workflow.yml:5]: Not until PRs are created. As far as I remember we opened to all branches to build workflows while working on the new implementation. Also, this aligns with other repos: https://github.com/Aiven-Open/s3-connector-for-apache-kafka/blob/main/.github/workflows/main_push_and_pull_request_workflow.yml

Let me know if I'm missing something, though.

Comment (AnatolyPopov) [reply .github/workflows/main_push_workflow.yml:5]: Ah, yeah, I remembered. But should we actually have 2 workflows or 1 with 2 conditions?

Comment (jeqo) [reply .github/workflows/main_push_workflow.yml:5]: We could actually. Was considering keeping them separate to publish images only (TODO) only on the main branch; but could be also done on PRs with a different tag 🤔 

If we prefer this, I'm happy to merge them

**Review by AnatolyPopov — APPROVED** [review]

**Pull request merged**